src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/features/form/validation/validationSlice.ts

```diff
@@ -0,0 +1,91 @@
+import { createAction, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { IComponentValidations, IValidations } from 'src/types';
+
+export interface IValidationState {
+  validations: IValidations;
+  invalidDataTypes: string[];
+  error: Error;
+  currentSingleFieldValidation: string;
+}
+
+export interface IUpdateComponentValidations {
+  layoutId: string;
+  validations: IComponentValidations;
+  componentId: string;
+  invalidDataTypes?: string[];
+}
+
+export interface IUpdateValidations {
+  validations: IValidations;
+}
+
+export interface IValidationActionRejected {
+  error: Error;
+}
+
+export interface ISetCurrentSingleFieldValidationAction {
+  dataModelBinding?: string;
+}
+
+const initialState: IValidationState = {
+  validations: {},
+  error: null,
+  invalidDataTypes: [],
+  currentSingleFieldValidation: null,
+};
+
+const moduleName = 'validation';
```

Comment:
If this is not used in more than one place it can be inlined.

Reply:
It's used in 2 places currently, and can then be used by any new actions that need to be defined for validation outside of the slice in the future.